models/login_source.go

```diff
@@ -394,7 +410,8 @@ func (s *LocalLoginSources) UpdateLoginSource(source *LoginSource) {
 	for i := range s.sources {
 		if s.sources[i].ID == source.ID {
 			*s.sources[i] = *source
-			break
+		} else if source.IsDefault {
+			s.sources[i].IsDefault = false
```

Comment:
This line actually has no effect theoretically because no DB updates in this method after this line.